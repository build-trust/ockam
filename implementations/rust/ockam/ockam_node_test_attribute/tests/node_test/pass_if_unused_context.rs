// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the function is passed a parameter of type `ockam::Context` but is unused.

#[ockam_node_test_attribute::node_test]
async fn my_test(_ctx: &mut ockam::Context) -> ockam::Result<()> {
    // _ctx.stop().unwrap();
    let _x = 42 as u8;
}
