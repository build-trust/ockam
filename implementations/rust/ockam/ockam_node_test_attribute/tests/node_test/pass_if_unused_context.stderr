error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/pass_if_unused_context.rs:4:1
  |
4 | / #[ockam_node_test_attribute::node_test]
5 | | async fn my_test(_ctx: &mut ockam::Context) -> ockam::Result<()> {
6 | |     // _ctx.stop().unwrap();
7 | |     let _x = 42 as u8;
8 | | }
  | |_^ consider adding a `main` function to `$DIR/tests/node_test/pass_if_unused_context.rs`
